lib/httpClient.js

```diff
@@ -249,20 +250,22 @@ function getMetricsOptions(metricsOptions = {}, url, metrics) {
 function decorateResponseWithDuration(axiosInstance) {
   axiosInstance.interceptors.response.use(
     (response) => {
-      response.config.metadata.endTime = new Date()
-      response.duration = response.config.metadata.endTime - response.config.metadata.startTime
+      const endTime = get(response, ['config', 'metadata', 'endTime'], new Date())
+      const startTime = get(response, ['config', 'metadata', 'startTime'])
+      response.duration = endTime - startTime
       return response
     },
     (error) => {
-      error.config.metadata.endTime = new Date()
-      error.duration = error.config.metadata.endTime - error.config.metadata.startTime
+      const endTime = get(error, ['config', 'metadata', 'endTime'], new Date())
```

Comment:
Why is it happening for config (or config.metadata) to be undefined? I wonder what the root cause is, because it feels to me that this pr is kind of hiding a problem rather than resolving it

Reply:
I receive this error when i use the httpClient.

```
{"level":50,"time":1734348316482,"pid":12368,"hostname":"NB198.local","reqId":"req-1","error":{"type":"Error","message":"Cannot read properties of undefined (reading
'metadata')","stack":"Error: Cannot read properties of undefined (reading 'metadata')\n    at HttpClient.makeCall (/Users/nb198/Documents/deploy-service/node_modules/@mia-platform/custom-plugin-lib/lib/httpClient.js:148:27)\n
  at processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at retrieveProvider (/Users/nb198/Documents/deploy-service/lib/get-provider-by-id.js:12:40)\n    at DeploymentSelector.deploymentHistoryFor
(/Users/nb198/Documents/deploy-service/lib/deployment-selector.js:325:73)\n    at Object.handler (/Users/nb198/Documents/deploy-service/lib/pipelines/deployment-history.js:226:9)"},"msg":"error on get provider from crud"}
./tests/integration.deployment-history.test.js 1> {"level":50,"time":1734348316482,"pid":12368,"hostname":"NB198.local","reqId":"req-1","error":{"type":"InternalServerError","message":"Error during provider
retrieval","stack":"InternalServerError: Error during provider retrieval\n    at /Users/nb198/Documents/deploy-service/lib/get-provider-by-id.js:17:11\n    at processTicksAndRejections
(node:internal/process/task_queues:105:5)\n    at retrieveProvider (/Users/nb198/Documents/deploy-service/lib/get-provider-by-id.js:12:40)\n    at DeploymentSelector.deploymentHistoryFor
(/Users/nb198/Documents/deploy-service/lib/deployment-selector.js:325:73)\n    at Object.handler
(/Users/nb198/Documents/deploy-service/lib/pipelines/deployment-history.js:226:9)","status":500,"statusCode":500,"expose":false},"msg":"failed deployment history retrieval"}
```